conda/common/logic.py

```diff
@@ -275,20 +275,41 @@ class Clauses(object):
     def __init__(self, m=0, sat_solver_cls=PycoSatSolver):
         self.names = {}
         self.indices = {}
-        self.unsat = False
-        self.m = m
-        self._sat_solver = sat_solver_cls()
-        # Bind some methods of _sat_solver to reduce lookups and call overhead.
-        self.add_clause = self._sat_solver.add_clause
-        self.add_clauses = self._sat_solver.add_clauses
+        self._clauses = _Clauses(m=m, sat_solver_cls=sat_solver_cls)
+
+    @property
+    def m(self):
+        return self._clauses.m
+
+    @property
+    def unsat(self):
+        return self._clauses.unsat
 
     def get_clause_count(self):
-        return self._sat_solver.get_clause_count()
+        return self._clauses.get_clause_count()
 
     def as_list(self):
-        return self._sat_solver.as_list()
+        return self._clauses.as_list()
+
+    def _check_variable(self, variable):
+        if 0 < abs(variable) <= self.m:
```

Comment:
These checks are new. They do not hurt performance-wise since they only happen on the interface level but not in the lower level performance critical code paths.